### PR TITLE
Add bellbot app on indigo branch

### PR DESCRIPTION
**Pull request by kunzel**
This moves the bellbot code into the strands_apps repository. The core parts of this code will remain here. The AAF dependent parts (ie the GUI) should be factored out and moved in the aaf_deployment repo (@yianni, can you please take care of this).

### CODE REVIEW

**Comment by marc-hanheide** [comment]
retest this please

**Comment by marc-hanheide** [comment]
The last comment was necessary to trigger jenkins, as up to https://github.com/strands-project/rosdistro/commit/885c8a62646e6491387ff55309eb61efc0455c15 the indigo build was still using `hydro-devel`, fixed now

**Comment by marc-hanheide** [comment]
jenkins says:
`Could not find robblog in keys`

Is robblog needed?

**Comment by gatsoulis** [comment]
From the discussions I thought that both server and gui should go to strands_apps.

Yianni

On 16 Feb 2015, at 23:16, Lars Kunze <notifications@github.com<mailto:notifications@github.com>> wrote:

This moves the bellbot code into the strands_apps repository. The core parts of this code will remain here. The AAF dependent parts (ie the GUI) should be factored out and moved in the aaf_deployment repo (@yiannihttps://github.com/yianni, can you please take care of this).

---

You can view, comment on, or merge this pull request online at:

  https://github.com/strands-project/strands_apps/pull/30

Commit Summary
-   Initial commit for the bellbot action server.
-   fixed several bugs, the bellbot can now patrol between points.
-   Added a client to test the server.
-   Added a service which can be used to set new targets for the next bellbot tour.
-   bellbot_gui working demo 1
-   Updated gitignore
-   Added launch file for the bellbot action server
-   Several fixes in the bellbot state machine.
-   Create README.md
-   Bellbot scheduler
-   latest changes after crash
-   added gui after crash
-   Merge remote-tracking branch 'lars/master'
-   introduced bellbot modes (1,2 working; 3 wip)
-   added new BellbotState
-   Merge remote-tracking branch 'lars/master'
-   updating bellbot schedule script
-   Bellbot message generation
-   Added setup.py
-   final state machine
-   gui stuff
-   bugfix
-   Merge pull request #1 from RaresAmbrus/master
-   mode 2 gui done
-   Schedulign gui
-   Merge branch 'master' of https://github.com/kunzel/bellbot_demo
-   updated launch file
-   Merge branch 'master' of https://github.com/RaresAmbrus/bellbot_demo
-   added shutdown state
-   on demand
-   Merge pull request #7 from RaresAmbrus/master
-   added categorical selection of destinations + other things
-   Merge branch 'master' of https://github.com/kunzel/bellbot_demo
-   add bellbot to strands_apps

File Changes
-   A bellbot_action_server/CMakeLists.txthttps://github.com/strands-project/strands_apps/pull/30/files#diff-0 (166)
-   A bellbot_action_server/action/bellbot.actionhttps://github.com/strands-project/strands_apps/pull/30/files#diff-1 (11)
-   A bellbot_action_server/launch/bellbot_action_server.launchhttps://github.com/strands-project/strands_apps/pull/30/files#diff-2 (13)
-   A bellbot_action_server/msg/BellbotState.msghttps://github.com/strands-project/strands_apps/pull/30/files#diff-3 (3)
-   A bellbot_action_server/package.xmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-4 (67)
-   A bellbot_action_server/src/agent.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-5 (42)
-   A bellbot_action_server/src/bellbot_server.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-6 (96)
-   A bellbot_action_server/src/bellbot_state_machine.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-7 (437)
-   A bellbot_action_server/src/bellbot_test_client.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-8 (57)
-   A bellbot_action_server/srv/Feedback.srvhttps://github.com/strands-project/strands_apps/pull/30/files#diff-9 (1)
-   A bellbot_action_server/srv/NewTarget.srvhttps://github.com/strands-project/strands_apps/pull/30/files#diff-10 (2)
-   A bellbot_gui/CMakeLists.txthttps://github.com/strands-project/strands_apps/pull/30/files#diff-11 (183)
-   A bellbot_gui/package.xmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-12 (59)
-   A bellbot_gui/scripts/foo.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-13 (52)
-   A bellbot_gui/setup.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-14 (13)
-   A bellbot_gui/src/**init**.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-15 (0)
-   A bellbot_gui/src/destination_data.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-16 (14)
-   A bellbot_gui/src/gui.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-17 (272)
-   A bellbot_gui/src/livescreen.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-18 (21)
-   A bellbot_gui/src/record_feedback.cpphttps://github.com/strands-project/strands_apps/pull/30/files#diff-19 (41)
-   A bellbot_gui/src/run.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-20 (16)
-   A bellbot_gui/srv/FeedbackDone.srvhttps://github.com/strands-project/strands_apps/pull/30/files#diff-21 (1)
-   A bellbot_gui/www/_htaccesshttps://github.com/strands-project/strands_apps/pull/30/files#diff-22 (3)
-   A bellbot_gui/www/cake.htmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-23 (8)
-   A bellbot_gui/www/cake.jpghttps://github.com/strands-project/strands_apps/pull/30/files#diff-24 (0)
-   A bellbot_gui/www/call_robot.htmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-25 (39)
-   A bellbot_gui/www/css/style.csshttps://github.com/strands-project/strands_apps/pull/30/files#diff-26 (178)
-   A bellbot_gui/www/feedback.htmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-27 (121)
-   A bellbot_gui/www/feedback.txthttps://github.com/strands-project/strands_apps/pull/30/files#diff-28 (1)
-   A bellbot_gui/www/filesave.phphttps://github.com/strands-project/strands_apps/pull/30/files#diff-29 (16)
-   A bellbot_gui/www/images/happy.pnghttps://github.com/strands-project/strands_apps/pull/30/files#diff-30 (0)
-   A bellbot_gui/www/images/neutral.pnghttps://github.com/strands-project/strands_apps/pull/30/files#diff-31 (0)
-   A bellbot_gui/www/images/sad.pnghttps://github.com/strands-project/strands_apps/pull/30/files#diff-32 (0)
-   A bellbot_gui/www/js/jquery-1.11.2.min.jshttps://github.com/strands-project/strands_apps/pull/30/files#diff-33 (4)
-   A bellbot_gui/www/js/script.jshttps://github.com/strands-project/strands_apps/pull/30/files#diff-34 (33)
-   A bellbot_gui/www/livescreen.htmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-35 (224)
-   A bellbot_gui/www/livescreen.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-36 (21)
-   A bellbot_gui/www/scheduling.htmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-37 (158)
-   A bellbot_scheduler/CMakeLists.txthttps://github.com/strands-project/strands_apps/pull/30/files#diff-38 (40)
-   A bellbot_scheduler/msg/BellbotScheduleRequest.msghttps://github.com/strands-project/strands_apps/pull/30/files#diff-39 (5)
-   A bellbot_scheduler/package.xmlhttps://github.com/strands-project/strands_apps/pull/30/files#diff-40 (30)
-   A bellbot_scheduler/scripts/bellbot_schedule_interface.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-41 (113)
-   A bellbot_scheduler/setup.pyhttps://github.com/strands-project/strands_apps/pull/30/files#diff-42 (11)

Patch Links:
-   https://github.com/strands-project/strands_apps/pull/30.patch
-   https://github.com/strands-project/strands_apps/pull/30.diff

**Comment by gatsoulis** [comment]
Don't think so, but not easy to check now as travelling. What's robblog? Will try to find out within the Bellbot team.

Yianni

On 17 Feb 2015, at 01:01, Marc Hanheide <notifications@github.com<mailto:notifications@github.com>> wrote:

jenkins says:
Could not find robblog in keys

Is robblog needed?

**Comment by kunzel** [comment]
@yianni, from your email I understood that there are some parts that depend on the AAF scenario. So, I agree that most of the GUI part can then stay in strands_apps. However, the depend parts should go in the aaf_deployment package. Maybe in form of a launch file if you can pass it dynamically!? 

@Jailander any idea what robblog is?

**Comment by marc-hanheide** [comment]
@kunzel robblog is here: https://github.com/strands-project/robblog I suppose it's a copy and paste error?

**Comment by kunzel** [comment]
@marc-hanheide, yes, I'd guess so.

**Comment by marc-hanheide** [comment]
@yianni said:

> Try commenting out last line on cmakelists which has some c dependencies.

I don't quite know if he'll be doing it or not... 

@yianni better comment on the pull request rather than replying to a jenkins email.

**Comment by marc-hanheide** [comment]
I don't see how commenting out the last line would help. Is it not just that a file is missing:

```
var/lib/jenkins/workspace/pr-indigo-strands_apps/monitored_vcs/bellbot_gui/src/record_feedback.cpp:3:38: fatal error: bellbot_gui/FeedbackDone.h: No such file or directory
 #include "bellbot_gui/FeedbackDone.h"
```

Ah, no, I'm wrong. This is a generate header (although just `std_srvs/Empty` could be used instead (see below). The it's the order of dependencies that is messed up here, I suppose.

File: bellbot_gui/srv/FeedbackDone.srv [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (marc-hanheide) [inline bellbot_gui/srv/FeedbackDone.srv:1]: Actually, there is already an Empty Service defined in std_srvs: http://wiki.ros.org/std_srvs so no need for this here.

**Comment by Jailander** [comment]
Hi @kunzel @yianni I think the best place for this is the AAF deployment repo https://github.com/strands-project/aaf_deployment I think we should move it there.

@marc-hanheide the problem is that the dependencies are not well defined and with the rush on Friday we commented out that line and its call on the code and probably the pr version still has those but are not important.

@kunzel @yianni shall we move this into aaf deployment repo and fix there?

**Comment by marc-hanheide** [comment]
@Jailander do you mean for all of this? We agreed that the actual "guiding" would be more general and should be in strands_apps. Do you disagree?

**Comment by marc-hanheide** [comment]
With @yianni being absent, can either @kunzel or @Jailander take care of fixing the issues?

**Comment by gatsoulis** [comment]
On 17 Feb 2015, at 12:25, Jaime Pulido Fentanes <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi @kunzelhttps://github.com/kunzel @yiannihttps://github.com/yianni I think the best place for this is the AAF deployment repo https://github.com/strands-project/aaf_deployment I think we should move it there.

In my opinion strands_apps is a more suitable place given the generic nature of the task, and in aaf repo the specific customisations. However, I am happy either way.

@marc-hanheidehttps://github.com/marc-hanheide the problem is that the dependencies are not well defined and with the rush on Friday we commented out that line and its call on the code and probably the pr version still has those but are not important.

Yes, pretty much this. It failed to compile when the packages were moved on Werner and commenting out this line seemed to fix it for no obvious reason.

@kunzelhttps://github.com/kunzel @yiannihttps://github.com/yianni shall we move this into aaf deployment repo and fix there?

As I said above either is fine places fine by me.  Can't do anything today as I am flying within the hour.

**Comment by Jailander** [comment]
So @marc-hanheide suggested leaving the bellbot scheduler and action server here and moving the GUI to AAF deployment. 
If no one opposes I'll do it as soon as I sort out some strands_navigation stuff

**Comment by Jailander** [comment]
Hi, moving bellbot_gui to a different repo #31 doesn't seem to be a good idea as it can create circular dependencies for releases, I fixed the compilation problem in #31 by removing FeedbackDone service type and making it a standard empty service. @kunzel can you cancel this PR as it is included in #31?

**Comment by kunzel** [comment]
Thanks @jailander

**Pull request closed without merging**